src/Modules/OrchardCore.Commerce/Serialization/ProductAttributeValueConverter.cs

```diff
@@ -0,0 +1,74 @@
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using OrchardCore.Commerce.Abstractions;
+using OrchardCore.Commerce.ProductAttributeValues;
+using System;
+using System.Collections.Generic;
+
+namespace OrchardCore.Commerce.Serialization;
+
+internal class ProductAttributeValueConverter : JsonConverter<IProductAttributeValue>
+{
+    private const string Type = "type";
+    private const string Value = "value";
+    private const string AttributeName = "attributename";
+
+    public override IProductAttributeValue ReadJson(
+        JsonReader reader,
+        Type objectType,
+        IProductAttributeValue existingValue,
+        bool hasExistingValue,
+        JsonSerializer serializer)
+    {
+        var attribute = (JObject)JToken.Load(reader);
+
+        var attributeName = attribute.Get<string>(AttributeName);
+        var typeName = attribute.Get<string>(Type);
+
+        return typeName switch
+        {
+            nameof(TextProductAttributeValue) => new TextProductAttributeValue(attributeName, attribute.Get<IEnumerable<string>>(Value)),
+            nameof(BooleanProductAttributeValue) => new BooleanProductAttributeValue(attributeName, attribute.Get<bool>(Value)),
+            nameof(NumericProductAttributeValue) => new NumericProductAttributeValue(attributeName, attribute.Get<decimal>(Value)),
+            _ => throw new InvalidOperationException($"Unknown or unsupported type \"{typeName}\"."),
+        };
+    }
+
+    public override void WriteJson(JsonWriter writer, IProductAttributeValue productAttributeValue, JsonSerializer serializer)
+    {
+        if (productAttributeValue is null)
+        {
+            return;
+        }
+
+        writer.WriteStartObject();
+
+        var typeName = productAttributeValue.GetType().Name;
+
+        writer.WritePropertyName(Type);
+        writer.WriteValue(typeName);
+
+        writer.WritePropertyName(Value);
+
+        if (typeName == nameof(TextProductAttributeValue))
```

Comment:
Instead of hard coding the type, it would be cleaner to check if `productAttributeValue.UntypedValue` is `IEnumerable<string>`, or even just `IEnumerable`.

Reply:
I've hit an unexpected case doing the same in CARE:

I was checking a property for its type and value, and when the type was derived from `IEnumerable` and the value was `null` I threw an `ArgumentError` asking the reader to assign an empty collection to the property when construction its containing class.

Turns out I forgot that `string : IEnumerable<char>`.

You might want to account for that, too.